tests/kernel/mem_protect/sys_sem/src/main.c

```diff
@@ -546,6 +555,10 @@ void test_sem_multiple_threads_wait(void)
 			     signal_count);
 
 		repeat_count++;
+
+		for (int i = 0; i < TOTAL_THREADS_WAITING; i++) {
+			k_thread_join(&multiple_tid[i], K_FOREVER);
```

Comment:
Great neat trick. Should we add example of usage to the Zephyr's docs, so users can know k_thread_join API can help to avoid race condition?

As I understand adding k_thread_join for the last function test_sem_multiple_threads_wait is made to be safe if a new test will be added. Because as I understand, for now that won't cause any race condition, because function executes in the last.